File: 1310 - Lucro.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int i, n, c, res, soma, receita[50];

    while(scanf("%d", &n) != EOF)
    {
        res = soma = 0;

        scanf("%d", &c);

        for (i = 0; i < n; i++)
        {
            scanf("%d", &receita[i]);
            soma += receita[i] - c;
            if (soma < 0)
                soma = 0;
            res = soma > res ? soma : res;
        }

        printf("%d\n", res);
    }
}

File: 1310 - Lucro.py
while True:
    try:
        n = int(input())

        res = soma = 0

        c = int(input())

        receita = []

        for i in range(n):
            receita.append(int(input()))
            soma += receita[i] - c
            if soma < 0:
                soma = 0
            res = soma if soma > res else res

        print(res)

    except:
        break